cli/asc.js

```diff
@@ -514,28 +517,35 @@ exports.main = function main(argv, options, callback) {
         tsNodeRegistered = true;
       }
       try {
-        const classOrModule = dynrequire(dynrequire.resolve(filename, { paths: [baseDir, process.cwd()] }));
-        if (typeof classOrModule === "function") {
-          Object.assign(classOrModule.prototype, {
-            program,
-            baseDir,
-            stdout,
-            stderr,
-            log: console.error,
-            readFile,
-            writeFile,
-            listFiles
-          });
-          transforms.push(new classOrModule());
-        } else {
-          transforms.push(classOrModule); // legacy module
-        }
+        transforms.push(dynrequire(dynrequire.resolve(filename, { paths: [baseDir, process.cwd()] })));
       } catch (e) {
         return callback(e);
       }
     }
   }
 
+  // Fix up the prototype of the transforms’ constructors and instantiate them.
+  try {
+    transforms = transforms.map(classOrModule => {
+      // Except if it’s a legacy module, just pass it through.
+      if (typeof classOrModule !== "function") {
+        return classOrModule; 
+      }
+      return Object.assign(classOrModule.prototype, {
+        program,
+        baseDir,
+        stdout,
+        stderr,
+        log: console.error,
+        readFile,
+        writeFile,
+        listFiles
+      });
+    });
+  } catch (e) {
+    return callback(e)
```

Comment:
```suggestion
    return callback(e);
```
last lint error it seems

Reply:
I need to check why I didn’t catch these locally...